meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor_git.bb

```
@@ -17,13 +19,16 @@ def mender_monitor_srcrev_from_src_uri(d, src_uri):
         src_uri_glob = src_uri_list[0][len("file://"):]
         # Get the filename
         filenames = glob.glob(src_uri_glob)
-        if len(filenames) != 1:
-            bb.error("Expected exactly one file, found: %s" % filenames)
-        filename = filenames[0]
-        # Now extract the git sha from the filename
-        m = re.match(r".*/mender-monitor-([0-9a-f]+)\.tar\.gz", filename)
+        if len(filenames) == 0:
+            bb.error("Failed to find mender monitor on path %s" % src_uri_glob)
+            bb.error("Please make sure SRC_URI_pn-mender-monitor is pointing to the downloaded tarball ")
+        elif len(filenames) != 1:
+            bb.warning("Expected exactly one file, found: %s" % filenames)
+        filename = os.path.basename(filenames[0])
+        # Now extract the SRCREV from the filename
+        m = re.match(r"mender-monitor-(.*)\.tar\.gz", filename)
```

I think this is too open. I think your fix is good intended (allow master) but we should try to be as strict as possible. I hence suggest:
```suggestion
        m = re.match(r"mender-monitor-(master|[0-9a-f]+)\.tar\.gz", filename)
```

---

If you really push it, we could even be strict in how long the git sha can be:
```suggestion
        m = re.match(r"mender-monitor-(master|[0-9a-f]7)\.tar\.gz", filename)
```